### DISCUSSION TITLE
No symlinks but: Error: Artifact could not be deployed. Please ensure the content does not contain any hard links, symlinks and total size is less than 10GB.

### DISCUSSION THREAD

**Discussion by NickBorgers** [community/community#40771] [Actions] [answered]
### Select Topic Area

Question

### Body

I can't figure out what I am doing to provoke failure of GitHub Pages deployment in an Action with the message:
```
Error: Artifact could not be deployed. Please ensure the content does not contain any hard links, symlinks and total size is less than 10GB.
```

[Example run which gets this behavior.](https://github.com/NickBorgers/node-red/actions/runs/3614560885)
As visible there [the artifact was created successfully, is of appropriate size (1.3MB), and contains no symlinks](https://github.com/NickBorgers/node-red/suites/9658646140/artifacts/462085716).

[I think I've followed the guidance published here for the deploy action.](https://github.com/actions/deploy-pages)
- [Creating artifact called `github-pages`](https://github.com/NickBorgers/node-red/blob/f9992aad61a9185ae5f32c1a0b310b1f3441a291/.github/workflows/publish.yml#L31-L35)
- [Have appropriate permissions for the action](https://github.com/NickBorgers/node-red/blob/f9992aad61a9185ae5f32c1a0b310b1f3441a291/.github/workflows/publish.yml#L10-L11)
- [Configured environment as recommended](https://github.com/NickBorgers/node-red/blob/f9992aad61a9185ae5f32c1a0b310b1f3441a291/.github/workflows/publish.yml#L21)
- Have allowed the branch to deploy to github-pages
<img width="1134" alt="image" src="https://user-images.githubusercontent.com/1139580/205509937-b101f68d-274a-4dc6-826c-416171237a82.png">

**Comment by ossanna16** [comment]
Hi there @NickBorgers and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by vitalykarasik** [comment]
Weird - I don't see "upload artifacts" action in a log.
I suggest to do these things:

- bump GH Actions versions for 'checkout' and 'upload-artifacts'
- add 'ls -l .automated-rendering/screenshot-capture/screenshots/' debug step before artifact uploading

**Reply by NickBorgers** [reply]
[The step right before the failed publish here is the upload-artifacts action](https://github.com/NickBorgers/node-red/actions/runs/3614560885/jobs/6091125256). Its debug log includes listing of the files that get included in the tar file.

[I can also see the files in the resulting artifact.](https://github.com/NickBorgers/node-red/suites/9658646140/artifacts/462085716)

[Regardless, switching to the pages-specific upload action fixed my problem.](https://github.com/NickBorgers/node-red/blob/d33e2e946b245a811e871da8ef26b41560cd0e5a/.github/workflows/publish.yml#L32)

**Comment by NickBorgers** [comment, marked as answer]
[Switching to `actions/upload-pages-artifact@v1` fixed my problem.](https://github.com/NickBorgers/node-red/blob/d33e2e946b245a811e871da8ef26b41560cd0e5a/.github/workflows/publish.yml#L32)

**Reply by mariari** [reply]
you probably got the idea for v3 from

https://github.com/actions/upload-artifact

which has a very similar name, I ran into the same issue.

**Reply by code4break** [reply]
I had the same issue. Thank you very much 👍

**Reply by naXa777** [reply]
Switching to [`actions/upload-pages-artifact@v2`](https://github.com/actions/upload-pages-artifact) fixed my problem.

**Reply by ajsobrino700** [reply]
The documentation of this actions is horrible. 
I have problem with both action.
Paste the output of the action with all information.

  with:
    name: github-pages
    path: _site/
    retention-days: 1
Run tar \
  tar \
    --dereference --hard-dereference \
    --directory "$INPUT_PATH" \
    -cvf "$RUNNER_TEMP/artifact.tar" \
    --exclude=.git \
    --exclude=.github \
    .
  shell: /usr/bin/sh -e {0}
  env:
    INPUT_PATH: _site/
tar: _site/: Cannot open: No such file or directory
tar: Error is not recoverable: exiting now
Error: Process completed with exit code 2.

**Reply by danielkutik** [reply]
i had a file with permission set to chmod 600... deleting that file/changing the permission to 644 resolved the issue for me

**Reply by shivaramr** [reply]
Hey @danielkutik , where did you delete that chmod 600?

**Reply by danielkutik** [reply]
you could do `chmod 755 path/to/problem/file` or `rm path/to/problem/file`

**Reply by TheDevelolper** [reply]
Would be great to see an example of your deployment file because I'm not sure where I should put the script

**Reply by laraibg786** [reply]
Instead of using `actions/upload-artifact`, use `actions/upload-pages-artifact` when uploading artifacts for the pages deployment. This works with `actions/deploy-pages@v4`
